File: Coding in Python/25th December 2025/Maximize Happiness of Selected Children.py
#Maximize Happiness of Selected Children
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort()
        result = 0
        c = 0
        i = len(happiness) - 1
        while i >= 0 and c < k:
            val = happiness[i] - c
            if val <= 0:
                break
            result += val
            c += 1
            i -= 1
        return result

File: coding/25th December 2025/Maximize Happiness of Selected Children.java
//Maximize Happiness of Selected Children
class Solution 
{
    public long maximumHappinessSum(int[] happiness, int k) 
    {
        Arrays.sort(happiness);
        int i = happiness.length - 1;
        long result = 0; 
        int c = 0;
        while(i >= 0 && c < k)
        {
            int val = (happiness[i] - c);
            if(val <= 0)
                break;
            result += val;
            c++;
            i--;
        }
        return result;
    }
}
